lib/chef/provider/package/apt.rb

```diff
@@ -229,6 +234,12 @@ def package_data_for(pkg)
           }
         end
 
+        def check_availability(name)
+          name.map do |pkg|
+            showpkg = run_noninteractive("apt-cache", "search", pkg).stdout
+            showpkg.empty? ? logger.warn("Unable to locate package  #{pkg} ") : pkg
+          end
```

Comment:
do you nee to call `.compact` here so you don't return a bunch of `nil`s?

Reply:
Thanks @jaymzh, Updated.